workshops/platformer/README.md

```diff
@@ -0,0 +1,717 @@
+
+```js
+if (keyDown(UP_ARROW)) {
+}
+```
+
+p5.play has this handy function [`keyDown()`](http://p5play.molleindustria.org/docs/classes/p5.play.html#method-keyDown) that will check if the up arrow key was pressed.
+
+To make the player jump, we'll modify its velocity in the y direction. p5.play has us covered, with this super straightforward `velocity` property, accessible by `player.velocity`. All we have to do is set it to `JUMP`. We'll do this within our conditional:
+
+```js
+if (keyDown(UP_ARROW)) {
+  player.velocity.y = JUMP;
+}
+```
+
+Save and refresh live preview, and give it a try by pressing your up arrow key!
+
+Oops, looks like we forgot to bring the player back down using gravity!
+
+```js
+if (keyDown(UP_ARROW)) {
+  player.velocity.y = JUMP;
+}
+player.velocity.y = player.velocity.y + GRAVITY;
+```
+
+That should do it. The player will definitely come back down now. Save and refresh to check it out.
+
+The player's going down, alright. In fact, it's going through the ground.
+
+We'll have to place a restriction on this. That is, we want to check if the player's y-position is greater than (flipped coordinate system) that of the surface of the ground. If so, we'd like to set the y-velocity back to 0, and ensure that the player rests on the ground.
+
+```js
+if (keyDown(UP_ARROW)) {
+  player.velocity.y = JUMP;
+}
+player.velocity.y = player.velocity.y + GRAVITY;
+
+if (Math.floor(player.position.y) > (height-50)-(player.height/2)) {
+  player.velocity.y = 0;
+  player.position.y = (height-50) - (player.height/2);
+}
+```
+
+Here, we perform a check to see if the player's y-position, rounded down to the nearest whole number, is greater than (i.e., dips below) the ground level. If that's indeed the case, then we stop moving it downwards, and stabilize the player at ground-level.
+
+## Part IV: Adding Obstacles
+
+### Generating Obstacles
+
+This isn't much of a game right now. All we can do is practice jumping.
+
+Let's add some obstacles to jump over.
+
+For this, we'll need to make another `Group`. Let's create one at the top, under our player declaration.
+
+```js
+var obstacleSprites;
+```
+
+Now we'll initialize it in `setup()`, in the same way we did `groundSprites`. Do this at the end of `setup()`, right after the creation of the player sprite:
+
+```js
+obstacleSprites = new Group();
+```
+
+Now that we've got somewhere to store the obstacles, we can start generating some within `draw()`.
+
+We can set the x-coordinate as `camera.position.x + width` to ensure they're generated off-screen to the right. We'll have the obstacles sit on the ground, by giving each a y-coordinate of `(height-50)-15`. Finally, we'll set the height and width of each obstacle at 30 for now.
+
+Let's type the following right above `drawSprites()`, inside `draw()`:
+
+```js
+var obstacle = createSprite(camera.position.x + width, (height-50) - 15, 30, 30);
+```
```

Comment:
Unless i'm misreading this, you haven't yet stated that you add var obstacle to our new Group() obstacleSprites, so this doesn't make too much sense. If i'm wrong I assume you mean that obstacle is generated in the same position repeatedly? Since it's in the draw function. If this is the case then there would be a ton of lag but no notable visual difference, so the user wouldn't be able to tell if there are multiple being generated.